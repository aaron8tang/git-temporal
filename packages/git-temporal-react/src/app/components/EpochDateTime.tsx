import React from 'react';

import { DateTime } from 'app/components/DateTime';

export interface EpochDateTimeProps {
  value: number;
  style?: object;
  displayOptions?: object;
}

// numbers for humans have commas
export const EpochDateTime = (props: EpochDateTimeProps): JSX.Element => {
  const dateTime = new Date(props.value * 1000);
  return (
    <DateTime
      value={dateTime}
      style={props.style}
      displayOptions={props.displayOptions}
    />
  );
};
