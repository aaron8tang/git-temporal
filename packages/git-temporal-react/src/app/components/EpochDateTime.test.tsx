import React from 'react';
import { shallow } from 'enzyme';

import { EpochDateTime } from './EpochDateTime';

describe('components/EpochDateTime', () => {
  describe('when rendered epoch time 1539369263 (Fri, 12 Oct 2018 18:34:23 GMT)', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(<EpochDateTime value={1539369263} />);
    });

    test('it should match snapshot (it should have rendered "3 weeks")', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
