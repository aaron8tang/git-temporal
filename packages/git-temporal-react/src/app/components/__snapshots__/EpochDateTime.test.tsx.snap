// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EpochDateTime when rendered epoch time 1539369263 (Fri, 12 Oct 2018 18:34:23 GMT) it should match snapshot (it should have rendered "3 weeks") 1`] = `
<DateTime
  value={2018-10-12T18:34:23.000Z}
/>
`;
